### PR TITLE
Don't reload sample from disk when reversing

### PR DESCRIPTION
**Pull request by DomClark**
Currently, when reversing a sample, `SampleBuffer` reloads it from disk. This is unnecessary and makes it impossible to do so in realtime code. This change still isn't realtime safe, since it locks a mutex, but a proper fix requires much wider changes, and it doesn't read from the disk any more at least.

Blocks #5695.

### CODE REVIEW

**Comment by Spekular** [comment]
Code looks reasonable to me! Do you want any testing on this? Anything specific?

**Comment by Spekular** [comment]
Wait, actually, doesn't this assome that every call to setReversed toggles the state? It appears to me that `_on` is meant to define if the sample should be reversed or not, so repeated calls to `setReversed` with the same argument should do nothing, right? Whereas here the sample would be reversed each time.

**Comment by DomClark** [comment]
Good catch! I was looking at similar code by Reflexe, but didn't quite take in that he changed the function from `setReversed(bool)` to `reverse()`. Should be fixed now. (Also, `m_reversed` is now set while the lock is held, which should be safer.)

In terms of testing, I don't think an awful lot is required. Flip some samples back and forth in AFP, make sure they save and load the same way round.

**Review by JohannesLorenz — APPROVED** [review]
LGTM

Still not RT safe, but a great performance increase.

**Comment by Spekular** [comment]
Did some very quick testing, everything worked as it should. I would probably have wrapped the whole function inside `if (m_reversed != on) { ... }` to avoid taking the locks if they're not needed, but in practice the `if` should always be true so it won't make any difference.

**Review by Spekular — APPROVED** [review]

**Comment by DomClark** [comment]
> I would probably have wrapped the whole function inside if (m_reversed != on) { ... } to avoid taking the locks if they're not needed

If this function is going to be called from multiple threads, the variable should be read and written under the same lock. Suppose two threads tried to reverse the sample simultaneously - both see that it's not reversed, so both reverse it and it ends up unreversed, even though `m_reversed` is now `true`.

Anyway, this has two approvals now, so I will merge it tomorrow if there are no objections.

**Comment by Spekular** [comment]
Ah, great point.

**Pull request merged**
